cmd/cloudFoundryCreateService.go

```diff
@@ -0,0 +1,103 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func cloudFoundryCreateService(config cloudFoundryCreateServiceOptions, telemetryData *telemetry.CustomData) {
+
+	cf := cloudfoundry.CFUtils{Exec: &command.Command{}}
+
+	var c = cf.Exec
+
+	//var c command.ExecRunner = &command.Command{}
+
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	var err, logouterr error
+
+	loginOptions := cloudfoundry.LoginOptions{
+		CfAPIEndpoint: config.CfAPIEndpoint,
+		CfOrg:         config.CfOrg,
+		CfSpace:       config.CfSpace,
+		Username:      config.Username,
+		Password:      config.Password,
+	}
+
+	err = cf.Login(loginOptions)
+	if err == nil {
+		err = runCloudFoundryCreateService(&config, telemetryData, c)
+	}
+	if err != nil {
+		logouterr = cf.Logout()
+		if logouterr != nil {
+			log.Entry().WithError(logouterr).Fatal("step execution failed")
```

Comment:
In case we have a `logoutError` we silently swallow the other error triggered e.g. from  `runCloudFoundryCreateService`. This makes it hard to do some troubleshooting.

Reply:
True. 
But, since there is logging in error cases directly in `cloudFoundryCreateServiceRequest` and federating/wrapping the error troubleshooting would seem okay to me. Could you further elaborate what could be changed in order to improve troubleshooting? Log-wise I have made some improvements and I think this is resolved due to the re-factoring.

Reply:
Thinks like this makes a huge difference, e.g. in case some less experienced operator has to troubleshoot. In case we have an error in `runCloudFoundryCreateService` and - at the same time - an error with the logout what should bubble up, the logout error or the issue with creating the service? When the logout error bubbles up that will be troubleshooted. In `java`/`groovy` there is an [explicit handling](https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getSuppressed()) for suppressed exceptions. Here we don't have that.

So I personally would log the logout error and let the other error bubble up. Anyway having good logging is good ...